kernel/kernel-api/src/main/java/io/delta/kernel/internal/metrics/SnapshotQueryContext.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (2024) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.delta.kernel.internal.metrics;
+
+import io.delta.kernel.metrics.SnapshotReport;
+import java.util.Optional;
+
+/** Stores the context for a given Snapshot query. Used to generate a {@link SnapshotReport} */
+public class SnapshotQueryContext {
+
+  public static SnapshotQueryContext forLatestSnapshot(String tablePath) {
+    return new SnapshotQueryContext(tablePath, Optional.empty(), Optional.empty());
+  }
+
+  public static SnapshotQueryContext forVersionSnapshot(String tablePath, long version) {
+    return new SnapshotQueryContext(tablePath, Optional.of(version), Optional.empty());
+  }
+
+  public static SnapshotQueryContext forTimestampSnapshot(String tablePath, long timestamp) {
+    return new SnapshotQueryContext(tablePath, Optional.empty(), Optional.of(timestamp));
+  }
+
+  private final String tablePath;
+  private Optional<Long> version;
```

Comment:
So I'm guessing this is not `providedVersion` because we expect this `version` variable to change ....

Could the constructor parameter be called `providedVersion`? Also worth some javadocs here for these variables. i.e. what does "provided" mean? (I know, but might as well add the context for future readers)

Reply:
Makes sense updated the param name & added docs